src/DotLiquid/Strainer.cs

```diff
@@ -15,6 +15,27 @@ static class DictionaryExtensions
                 return dic[key] = factory();
             return found;
         }
+
+    }
+
+    static class MethodInfoExtensions
+    {
+        public static int GetNonContextParameterCount(this MethodInfo method)

```

Comment:
Add XML comments for public classes and public methods.

Reply:
And yes we do have existing code that violates this, not particularly easy to decipher.